### DISCUSSION TITLE
RSS feed

### DISCUSSION THREAD

**Discussion by austintoddj** [austintoddj/canvas#923] [New Features]
The `feature/feed` branch in the project contains the necessary classes to accomplish this. There's one problem though:

Canvas doesn't _know_ about how posts are routed from the frontend. The link value in each `<entry>` in the feed itself needs to be an absolute path to where the post can be publicly viewed.

So, the only viable solution I can see is passing some sort of config variable to serve as the path prefix.

That would make it so the following scenarios would be covered:

- Jim structures his frontend routes like `www.athlead.com/posts/post-slug`
- Pam structures her frontend routes like `www.theartinstitute.com/post-slug`

However, Dwight wants his to look more personalized to himself:

- He structures his routes like `www.schrutefarms.com/dwight/post-slug`
- And his brother also blogs on his platform at `www.schrutefarms.com/mose/post-slug`

When it's a dynamic routing structure, even a config variable passed to Canvas won't do the job.

I'd really like to see this feature as a part of the package, but until I find a solution that will work for all these scenarios, and more, it's stuck where it is.

Related issues: #686, #490

**Comment by paulhennell** [comment]
Not sure where this got but doesn't look like this exists yet. 

I was looking for a way to get the url from a $post, and was expecting a callable to pass in somewhere, which would suit the problems mentioned above. 

My instinct, was using laravel macros in the CanvasServiceProvider to provide a `$post->url()` method that can use whatever url stratergy is desired:

```
Post::macro('url', function (){
            return route('post.show', ['slug'=>$this->slug]);
          //  return url($this->slug);
          //  return "/$this->user->name/$this->slug";
          //etc
});
````
Turns out models arn't macroable, so that doesn't work, but you could do something similar.  Maybe a default `$post->url()` that tries to build the string from config settings, or user sets `Post::setUrlMethod($callable);` in the service provider and gives thier own way to resolve it.

**Reply by austintoddj** [reply]
You're right @paulhennell, it doesn't exist yet. I haven't really made use of macros before, but the solution seems pretty straightforward. Would you care to make a PR to the `develop` branch with your idea?

**Comment by austintoddj** [comment]
Closing this out with [Canvas v7.0.0](https://github.com/austintoddj/canvas/releases/tag/v7.0.0).

Optional `canvas:ui` now includes an RSS feed at `/canvas-ui/feed` (`canvas-ui.feed`), with absolute post links derived from the published reader routes — which solves the “Canvas doesn’t know frontend URLs” problem for the starter frontend.

```bash
php artisan canvas:ui

If you build your own reader, reuse the same pattern (named route → absolute URL in the feed). Host apps can also listen for post lifecycle events / webhooks if they need custom syndication.
